runtime/interpreter/interpreter.go

```diff
@@ -823,40 +823,50 @@ func (interpreter *Interpreter) resultValue(returnValue Value, returnType sema.T
 	return NewEphemeralReferenceValue(interpreter, false, returnValue, returnType)
 }
 
-func (interpreter *Interpreter) visitConditions(conditions []*ast.Condition) {
+func (interpreter *Interpreter) visitConditions(conditions ast.Conditions, kind ast.ConditionKind) {
 	for _, condition := range conditions {
-		interpreter.visitCondition(condition)
+		interpreter.visitCondition(condition, kind)
 	}
 }
 
-func (interpreter *Interpreter) visitCondition(condition *ast.Condition) {
+func (interpreter *Interpreter) visitCondition(condition ast.Condition, kind ast.ConditionKind) {
 
-	// Evaluate the condition as a statement, so we get position information in case of an error
+	switch condition := condition.(type) {
+	case *ast.TestCondition:
+		// Evaluate the condition as a statement, so we get position information in case of an error
+		statement := ast.NewExpressionStatement(interpreter, condition.Test)
 
-	statement := ast.NewExpressionStatement(interpreter, condition.Test)
+		result, ok := interpreter.evalStatement(statement).(ExpressionResult)
 
-	result, ok := interpreter.evalStatement(statement).(ExpressionResult)
+		value, valueOk := result.Value.(BoolValue)
```

Comment:
If the above casting failed, `result` would be `nil` here

Reply:
Just kept the code as-is, but I get your point.

Both type assertions must always succeed (the checker ensures it), so we can make the failures unreachable cases.

Refactored in 835bd6d53e66c9dde66a6aef11d68d0fbfe79522